### ISSUE TITLE
Registering Devices via REST

### ISSUE THREAD

**Issue by agtzdimi** [linksmart/device-gateway#40]
Hello,

Thank you for your work and the projects :) 
I have a question regarding the device-gateway.

My requirement is to register devices and keep them alive through the device-gateway on the fly. In this way while the service-catalog and device-gateway are running I can create separate MQTT topics to subscribe and publish.

For example: 
Currently it is possible to configure having an array of resources: 

`{
  "name": "MQTTSwitch",
  "description": "MQTT test switch service",
  "meta": {},
  "ttl": 120,
  "resources": [
    {
      "type": "Resource",
      "name": "Switch",
      "agent": {
        "type": "service",
        "dir": null,
        "exec": "agent-examples/mqtt-switch/switch.sh"
      },
      "representation": {},
      "protocols": [
        {
          "type": "MQTT",
          "methods": [
            "PUB",
            "SUB"
          ],
          "content-types": [
            "text/plain"
          ],
          "pub_topic": "/test/switch/pub",
          "sub_topic": "/test/switch/sub"
        },
        {
          "type": "REST",
          "methods": [
            "GET",
            "POST",
            "PUT"
          ],
          "content-types": [
            "text/plain"
          ]
        }
      ]
    },
    {
      "type": "Resource",
      "name": "Switch2",
      "agent": {
        "type": "service",
        "dir": null,
        "exec": "agent-examples/mqtt-switch/switch.sh"
      },
      "representation": {},
      "protocols": [
        {
          "type": "MQTT",
          "methods": [
            "PUB",
            "SUB"
          ],
          "content-types": [
            "text/plain"
          ],
          "pub_topic": "/test/switch/pub2",
          "sub_topic": "/test/switch/sub2"
        },
        {
          "type": "REST",
          "methods": [
            "GET",
            "POST",
            "PUT"
          ],
          "content-types": [
            "text/plain"
          ]
        }
      ]
    }
  ]
}
`

Is it possible to add multiple other resources entries after the start of the device-gateway?

**Comment by farshidtz** [comment]
Hi there, 

I'm glad that you find this project interesting.

Resource configuration is currently possible only using config files. Dynamic configuration of devices has been a long lasting feature request (#34) but hasn't been added yet. We'll prioritize this issue and implement it in the following weeks.

For the time being, you can update the configuration file and restart device-gateway. It restarts very quickly and you won't lose MQTT messages because the QoS is set to 1.

**Comment by agtzdimi** [comment]
Thank you for your quick reply.

Another thing I would like to ask is about the publish topic. Is it feasible to publish the message to another IP rather than localhost?

**Comment by farshidtz** [comment]
Yes of course, this is set in the [main configuration file](https://github.com/linksmart/device-gateway/blob/master/sample_conf/device-gateway.json) or retrieved from the Service Catalog.

**Comment by agtzdimi** [comment]
Well, my point is to have for example multiple resources, subscribing to 'localhost' and the first publish to 'IP1', the second one to 'IP2' etc. If I am correct the main config file is referring to one global MQTT broker right?

**Comment by farshidtz** [comment]
No, that's currently not possible but I can imagine the usefulness. Could you make a separate issue / feature request for that?

**Comment by agtzdimi** [comment]
Of course I will do it :) 

Thank you
